### ISSUE TITLE
[v2] test failure `TestDevSyncDefaultNamespace`

### ISSUE THREAD

**Issue by tejal29** [GoogleContainerTools/skaffold#7286]
Failure logs 

```
=== RUN   TestDevSyncDefaultNamespace/manual_sync_with_polling_trigger
    helper.go:254: Running [skaffold build -f skaffold-manual.yaml --default-repo gcr.io/k8s-skaffold] in testdata/file-sync
    helper.go:254: Ran [skaffold build -f skaffold-manual.yaml --default-repo gcr.io/k8s-skaffold] in 4.15 seconds
    sync_test.go:94: Running [skaffold dev -f skaffold-manual.yaml --default-repo gcr.io/k8s-skaffold --trigger polling] in testdata/file-sync
time="2022-04-11T22:52:43Z" level=info msg="Waiting for pods [test-file-sync] to be ready" subtask=-1 task=DevLoop
time="2022-04-11T22:52:43Z" level=warning msg="k8s/*.yaml did not match any file" subtask=-1 task=DevLoop
time="2022-04-11T22:52:45Z" level=error msg="WARNING: Could not open the configuration file: [/home/runner/.config/gcloud/configurations/config_default]."
time="2022-04-11T22:52:45Z" level=error msg="ERROR: (gcloud.config.config-helper) You do not currently have an active account selected."
time="2022-04-11T22:52:45Z" level=error msg="Please run:"
time="2022-04-11T22:52:45Z" level=error
time="2022-04-11T22:52:45Z" level=error msg="  $ gcloud auth login"
time="2022-04-11T22:52:45Z" level=error
time="2022-04-11T22:52:45Z" level=error msg="to obtain new credentials."
time="2022-04-11T22:52:45Z" level=error
time="2022-04-11T22:52:45Z" level=error msg="If you have already logged in with a different account:"
time="2022-04-11T22:52:45Z" level=error
time="2022-04-11T22:52:45Z" level=error msg="    $ gcloud config set account ACCOUNT"
time="2022-04-11T22:52:45Z" level=error
time="2022-04-11T22:52:45Z" level=error msg="to select an already authenticated account to use."
failed to resolve the digest of gcr.io/k8s-skaffold/test-file-sync:ac094d1e06bdb107b34b722ed349fbabe93d28885c797e611d2d4ae3acea5d55: does the image exist remotely?
    helper.go:240: Ran [skaffold dev -f skaffold-manual.yaml --default-repo gcr.io/k8s-skaffold --trigger polling] in 2.723 seconds
    stackdump.go:38: triggering skaffold stacktrace request
time="2022-04-11T22:57:43Z" level=warning msg="[df -h]" subtask=-1 task=DevLoop
Filesystem      Size  Used Avail Use% Mounted on
/dev/root        84G   61G   23G  74% /
devtmpfs        3.4G     0  3.4G   0% /dev
tmpfs           3.4G  4.0K  3.4G   1% /dev/shm
tmpfs           695M  1.2M  694M   1% /run
tmpfs           5.0M     0  5.0M   0% /run/lock
tmpfs           3.4G     0  3.4G   0% /sys/fs/cgroup
/dev/loop0       62M   62M     0 100% /snap/core20/1405
/dev/loop1       45M   45M     0 100% /snap/snapd/15314
/dev/loop2       44M   44M     0 100% /snap/snapd/15177
/dev/sda15      105M  5.2M  100M   5% /boot/efi
/dev/loop3       68M   68M     0 100% /snap/lxd/22753
/dev/sdb1        14G  4.1G  9.0G  32% /mnt

time="2022-04-11T22:57:43Z" level=warning msg="[kubectl -n default get pods -oyaml]" subtask=-1 task=DevLoop
apiVersion: v1
items: []
kind: List
metadata:
  resourceVersion: ""
  selfLink: ""

time="2022-04-11T22:57:43Z" level=warning msg="[kubectl -n default logs pod/test-file-sync]" subtask=-1 task=DevLoop
Error from server (NotFound): pods "test-file-sync" not found

    util.go:241: Timed out waiting for pods [test-file-sync] in namespace "default"
    panic.go:642: Running [skaffold delete -f skaffold-manual.yaml --default-repo gcr.io/k8s-skaffold] in testdata/file-sync
    helper.go:198: Skaffold log:
         
    helper.go:240: Ran [skaffold delete -f skaffold-manual.yaml --default-repo gcr.io/k8s-skaffold] in 497.004µs
    helper.go:198: Skaffold log:
         Listing files to watch...
        >  - test-file-sync
        > Generating tags...
        >  - test-file-sync -> gcr.io/k8s-skaffold/test-file-sync:v1.38.0-87-g7e7a1d1af
        > Checking cache...
        >  - test-file-sync: Not found. Building
        > Starting build...
        > Found [minikube] context, using local docker daemon.
        > Building [test-file-sync]...
        > Target platforms: [linux/amd64]
        > #1 [internal] load build definition from Dockerfile
        > #1 sha256:a021ccd1ff8f18388f9a83e3bd4473322d02058c3b9cf6705fb691d2ebb1d562
        > #1 transferring dockerfile: 207B done
        > #1 DONE 0.0s
        > 
        > #2 [internal] load .dockerignore
        > #2 sha256:8dac780e8d45840f435d77245190fd809200a069c97eeb4fcfa6e087a610bb0e
        > #2 transferring context: 2B done
        > #2 DONE 0.0s
        > 
        > #3 [internal] load metadata for docker.io/library/busybox:latest
        > #3 sha256:da853382a7535e068feae4d80bdd0ad2567df3d5cd484fd68f919294d091b053
        > #3 DONE 0.0s
        > 
        > #4 [1/3] FROM docker.io/library/busybox
        > #4 sha256:08a03f3ffe5fba421a6403c31e153425ced631d108868f30e04985f99d69326e
        > #4 DONE 0.0s
        > 
        > #5 [internal] load build context
        > #5 sha256:bfa65336cd7d21c91fcda4b2f3a61ecd26b4bbe11ae67eda438e19442788e65d
        > #5 transferring context: 24B done
        > #5 DONE 0.0s
        > 
        > #6 [2/3] COPY foo /
        > #6 sha256:009c0369fa22c4f90f7edaee4a58cb9d08d06ec99692cb5c2c21dd0aaf1b8e9a
        > #6 DONE 0.0s
        > 
        > #7 [3/3] RUN echo "bar" > /foo
        > #7 sha256:bc0ac7a6f8d59c57bfb64228b98e1894c410c9bd0505c4ed5bebbce088f42881
        > #7 DONE 0.3s
        > 
        > #8 exporting to image
        > #8 sha256:e8c613e07b0b7ff33893b694f7759a10d42e180f2b4dc349fb57dc6b71dcab00
        > #8 exporting layers
        > #8 exporting layers 0.9s done
        > #8 writing image sha256:ac094d1e06bdb107b34b722ed349fbabe93d28885c797e611d2d4ae3acea5d55 done
        > #8 naming to gcr.io/k8s-skaffold/test-file-sync:v1.38.0-87-g7e7a1d1af done
        > #8 DONE 0.9s
        > Build [test-file-sync] succeeded
        > 
=== RUN   TestDevSyncDefaultNamespace/manual_sync_with_notify_trigger
    helper.go:254: Running [skaffold build -f skaffold-manual.yaml --default-repo gcr.io/k8s-skaffold] in testdata/file-sync
    helper.go:254: Ran [skaffold build -f skaffold-manual.yaml --default-repo gcr.io/k8s-skaffold] in 622.099[826](https://github.com/GoogleContainerTools/skaffold/runs/5980937660?check_suite_focus=true#step:16:826)ms
    sync_test.go:94: Running [skaffold dev -f skaffold-manual.yaml --default-repo gcr.io/k8s-skaffold --trigger notify] in testdata/file-sync
time="2022-04-11T22:57:43Z" level=info msg="Waiting for pods [test-file-sync] to be ready" subtask=-1 task=DevLoop
time="2022-04-11T22:57:44Z" level=warning msg="k8s/*.yaml did not match any file" subtask=-1 task=DevLoop
time="2022-04-11T22:57:44Z" level=error msg="WARNING: Could not open the configuration file: [/home/runner/.config/gcloud/configurations/config_default]."
time="2022-04-11T22:57:44Z" level=error msg="ERROR: (gcloud.config.config-helper) You do not currently have an active account selected."
time="2022-04-11T22:57:44Z" level=error msg="Please run:"
time="2022-04-11T22:57:44Z" level=error
time="2022-04-11T22:57:44Z" level=error msg="  $ gcloud auth login"
time="2022-04-11T22:57:44Z" level=error
time="2022-04-11T22:57:44Z" level=error msg="to obtain new credentials."
time="2022-04-11T22:57:44Z" level=error
time="2022-04-11T22:57:44Z" level=error msg="If you have already logged in with a different account:"
time="2022-04-11T22:57:44Z" level=error
time="2022-04-11T22:57:44Z" level=error msg="    $ gcloud config set account ACCOUNT"
time="2022-04-11T22:57:44Z" level=error
time="2022-04-11T22:57:44Z" level=error msg="to select an already authenticated account to use."
failed to resolve the digest of gcr.io/k8s-skaffold/test-file-sync:ac094d1e06bdb107b34b722ed349fbabe93d28[885](https://github.com/GoogleContainerTools/skaffold/runs/5980937660?check_suite_focus=true#step:16:885)c797e611d2d4ae3acea5d55: does the image exist remotely?
    helper.go:240: Ran [skaffold dev -f skaffold-manual.yaml --default-repo gcr.io/k8s-skaffold --trigger notify] in 1.248 second
    stackdump.go:38: triggering skaffold stacktrace request
time="2022-04-11T23:02:43Z" level=warning msg="[df -h]" subtask=-1 task=DevLoop
Filesystem      Size  Used Avail Use% Mounted on
/dev/root        84G   61G   23G  74% /
devtmpfs        3.4G     0  3.4G   0% /dev
tmpfs           3.4G  4.0K  3.4G   1% /dev/shm
tmpfs           695M  1.2M  694M   1% /run
tmpfs           5.0M     0  5.0M   0% /run/lock
tmpfs           3.4G     0  3.4G   0% /sys/fs/cgroup
/dev/loop0       62M   62M     0 100% /snap/core20/1405
/dev/loop1       45M   45M     0 100% /snap/snapd/15314
/dev/loop2       44M   44M     0 100% /snap/snapd/15177
/dev/sda15      105M  5.2M  100M   5% /boot/efi
/dev/loop3       68M   68M     0 100% /snap/lxd/22753
/dev/sdb1        14G  4.1G  9.0G  32% /mnt

time="2022-04-11T23:02:43Z" level=warning msg="[kubectl -n default get pods -oyaml]" subtask=-1 task=DevLoop
apiVersion: v1
items: []
kind: List
metadata:
  resourceVersion: ""
  selfLink: ""

time="2022-04-11T23:02:43Z" level=warning msg="[kubectl -n default logs pod/test-file-sync]" subtask=-1 task=DevLoop
Error from server (NotFound): pods "test-file-sync" not found

    util.go:241: Timed out waiting for pods [test-file-sync] in namespace "default"
    panic.go:642: Running [skaffold delete -f skaffold-manual.yaml --default-repo gcr.io/k8s-skaffold] in testdata/file-sync
    helper.go:198: Skaffold log:
         
    helper.go:198: Skaffold log:
         Listing files to watch...
        >  - test-file-sync
        > Generating tags...
        >  - test-file-sync -> gcr.io/k8s-skaffold/test-file-sync:v1.38.0-87-g7e7a1d1af
        > Checking cache...
        >  - test-file-sync: Found. Tagging
        > 
=== RUN   TestDevSyncDefaultNamespace/inferred_sync_with_notify_trigger
    helper.go:254: Running [skaffold build -f skaffold-infer.yaml --default-repo gcr.io/k8s-skaffold] in testdata/file-sync
    helper.go:254: Ran [skaffold build -f skaffold-infer.yaml --default-repo gcr.io/k8s-skaffold] in 628.789372ms
    sync_test.go:94: Running [skaffold dev -f skaffold-infer.yaml --default-repo gcr.io/k8s-skaffold --trigger notify] in testdata/file-sync
time="2022-04-11T23:02:44Z" level=info msg="Waiting for pods [test-file-sync] to be ready" subtask=-1 task=DevLoop
time="2022-04-11T23:02:45Z" level=warning msg="k8s/*.yaml did not match any file" subtask=-1 task=DevLoop
time="2022-04-11T23:02:45Z" level=error msg="WARNING: Could not open the configuration file: [/home/runner/.config/gcloud/configurations/config_default]."
time="2022-04-11T23:02:45Z" level=error msg="ERROR: (gcloud.config.config-helper) You do not currently have an active account selected."
time="2022-04-11T23:02:45Z" level=error msg="Please run:"
time="2022-04-11T23:02:45Z" level=error
time="2022-04-11T23:02:45Z" level=error msg="  $ gcloud auth login"
time="2022-04-11T23:02:45Z" level=error
time="2022-04-11T23:02:45Z" level=error msg="to obtain new credentials."
time="2022-04-11T23:02:45Z" level=error
time="2022-04-11T23:02:45Z" level=error msg="If you have already logged in with a different account:"
time="2022-04-11T23:02:45Z" level=error
time="2022-04-11T23:02:45Z" level=error msg="    $ gcloud config set account ACCOUNT"
time="2022-04-11T23:02:45Z" level=error
time="2022-04-11T23:02:45Z" level=error msg="to select an already authenticated account to use."
failed to resolve the digest of gcr.io/k8s-skaffold/test-file-sync:ac094d1e06bdb107b34b722ed349fbabe93d2[888](https://github.com/GoogleContainerTools/skaffold/runs/5980937660?check_suite_focus=true#step:16:888)5c797e611d2d4ae3acea5d55: does the image exist remotely?
    helper.go:240: Ran [skaffold dev -f skaffold-infer.yaml --default-repo gcr.io/k8s-skaffold --trigger notify] in 1.306 second
    stackdump.go:38: triggering skaffold stacktrace request
time="2022-04-11T23:07:44Z" level=warning msg="[df -h]" subtask=-1 task=DevLoop
Filesystem      Size  Used Avail Use% Mounted on
/dev/root        84G   61G   23G  74% /
devtmpfs        3.4G     0  3.4G   0% /dev
tmpfs           3.4G  4.0K  3.4G   1% /dev/shm
tmpfs           695M  1.2M  694M   1% /run
tmpfs           5.0M     0  5.0M   0% /run/lock
tmpfs           3.4G     0  3.4G   0% /sys/fs/cgroup
/dev/loop0       62M   62M     0 100% /snap/core20/1405
/dev/loop1       45M   45M     0 100% /snap/snapd/15314
/dev/loop2       44M   44M     0 100% /snap/snapd/15177
/dev/sda15      105M  5.2M  100M   5% /boot/efi
/dev/loop3       68M   68M     0 100% /snap/lxd/22753
/dev/sdb1        14G  4.1G  9.0G  32% /mnt

time="2022-04-11T23:07:44Z" level=warning msg="[kubectl -n default get pods -oyaml]" subtask=-1 task=DevLoop
apiVersion: v1
items: []
kind: List
metadata:
  resourceVersion: ""
  selfLink: ""

time="2022-04-11T23:07:44Z" level=warning msg="[kubectl -n default logs pod/test-file-sync]" subtask=-1 task=DevLoop
Error from server (NotFound): pods "test-file-sync" not found

    util.go:241: Timed out waiting for pods [test-file-sync] in namespace "default"
    panic.go:642: Running [skaffold delete -f skaffold-infer.yaml --default-repo gcr.io/k8s-skaffold] in testdata/file-sync
    helper.go:198: Skaffold log:
         
    helper.go:198: Skaffold log:
         Listing files to watch...
        >  - test-file-sync
        > Generating tags...
        >  - test-file-sync -> gcr.io/k8s-skaffold/test-file-sync:v1.38.0-87-g7e7a1d1af
        > Checking cache...
        >  - test-file-sync: Found. Tagging
        >

```

**Comment by ericzzzzzzz** [comment]
TL;DR The issue posted seems resolved, but there is another issue that blocks us from un-skipping the test. Looks like render in v2 is not completed, we can recover the test along with [TestDevSync and TestDevAutoSync](https://github.com/GoogleContainerTools/skaffold/blob/main/integration/sync_test.go#L57-L115)

The test was supported to run with minikube context, but the log posted indicates that the test was running with a gke context which caused it failing. 
This seemed like a CI agent issue:
 - It shouldn't use gke kubernetes context, even it uses if should be authenticated. 
 - The error log can be reproduced locally by switching kuberntetes context to gke cluster and run ``gcloud auth revoke`` to logout session, not sure what's happening on the CI machine at that time. The wrong context issue seems resolved by itself, as the log for that test shows this: 

```bash
== RUN   TestDevSyncDefaultNamespace
=== RUN   TestDevSyncDefaultNamespace/manual_sync_with_polling_trigger
    helper.go:259: Running [skaffold build -f skaffold-manual.yaml --default-repo gcr.io/k8s-skaffold] in testdata/file-sync
    helper.go:259: Ran [skaffold build -f skaffold-manual.yaml --default-repo gcr.io/k8s-skaffold] in 3.39 seconds
    sync_test.go:97: Running [skaffold dev -f skaffold-manual.yaml --default-repo gcr.io/k8s-skaffold --trigger polling] in testdata/file-sync
time="2022-05-05T19:52:18Z" level=info msg="Waiting for pods [test-file-sync] to be ready" subtask=-1 task=DevLoop
time="2022-05-05T19:52:18Z" level=info msg="Still waiting for pods [test-file-sync]" subtask=-1 task=DevLoop
time="2022-05-05T19:52:18Z" level=info msg="Still waiting for pods [test-file-sync]" subtask=-1 task=DevLoop
time="2022-05-05T19:52:18Z" level=info msg="Still waiting for pods [test-file-sync]" subtask=-1 task=DevLoop
time="2022-05-05T19:52:18Z" level=warning msg="k8s/*.yaml did not match any file" subtask=-1 task=DevLoop
time="2022-05-05T19:52:20Z" level=warning msg="k8s/*.yaml did not match any file" subtask=-1 task=DevLoop
nothing to deploy
    helper.go:245: Ran [skaffold dev -f skaffold-manual.yaml --default-repo gcr.io/k8s-skaffold --trigger polling] in 2.049 seconds
    stackdump.go:38: triggering skaffold stacktrace request
```
This integration test doesn't have authentication issue, but test is still failing due to ``nothing to deploy`` error. Note that there are 2 warning logs show  k8s/*.yaml did not match any file, however skaffold should use
```yaml
deploy:
 kubectl:
   manifests:
     - pod.yaml
```
for deployment,   not sure why it still looks into ``k8s/*.yaml`` , seems something broken. creating a k8s folder and move pod.yaml and changing manifests to k8s/pod.yaml can make the test pass, but I would suggest not to do that as the workaround is just to cover a problem not fixing the issue. I would like to wait v2 render complete then un-skip this test.
